Tests/Reader/Makefile

```diff
@@ -0,0 +1,6 @@
+ALL: compile run
+compile: reader.cpp main.cpp
+	g++ -o compile reader.cpp main.cpp -I.
+
+run: reader.cpp main.cpp
+	./compile "converted_ADM_stellartable.dat"
```

Comment:
The targets here are wrong. You want to only run after the reader has been compiled so it should look like this:
```
run: reader
	./reader "converted_ADM_stellartable.dat"
```

Reply:
It's also good practice to add two special targets to your make file.

The first is `clean` which removes build artifacts:
```
clean:
	$(RM) reader reader.o main.o
```
where $(RM) resolves to `rm -f` by default.

and 
```
.PHONY: all run clean
```
which tells gnu make that the `all`, `run`, and `clean` targets don't actually generate files they're just steps in the build process.

Reply:
One more thing, you can assign variables just like a shell script. So you can for example specify 
```
CC:=g++
```
to specify the compiler should be `g++` and then use `$(CC)` everywhere you would use `g++`. This is often good practice to make your makefile more flexible.